### PR TITLE
Add Asset Registries tile to public dashboard (12240)

### PR DESCRIPTION
**Pull request by dimak1**
Issue #: https://github.com/bcgov/entity/issues/12240

Description:

Display different tiles for public users based on their product:
- PPR - 'My Personal Property Registry' tile
- MHR Product - 'My Manufactured Home Registry' tile
- PPR and MHR Products - 'My Asset Registries' tile

Screenshots

**PPR**

Screen Shot 2022-05-31 at 8 08 43 AM

**MHR and PPR**

Screen Shot 2022-05-31 at 3 49 59 PM

### CODE REVIEW

File: components/UserProduct.vue
Comment (cameron-freshworks) [inline components/UserProduct.vue:62]: Looks like a pretty generic class, does this affect the images in ALL the tiles? 
Is that requested by design?

File: enums/productCode.ts
Comment (cameron-freshworks) [inline enums/productCode.ts:6]: o interesting, there is a code explicitly for having both product subscriptions?

File: utils/user-product-helper.ts
Comment (cameron-freshworks) [inline utils/user-product-helper.ts:93]: I like the approach but is it required to create a unique code explicit to having both roles?
Would a simple computed value that checks for both roles suffice?

I feel like you could avoid having to add a new code and the creation of it. 

something like..
`['ppr', 'mhr'].every(role => { return products.includes(role) })`

File: components/UserProduct.vue
Comment (dimak1) [reply components/UserProduct.vue:62]: Yes, there is a design requirement to make tiles smaller, including the images.

Screen Shot 2022-05-31 at 8 34 34 AM

**Review by doug-lovett — APPROVED** [review]
No comments in addition to Cameron's. In future, Severin can also be added as a reviewer: he has done a lot of work on this site.

**Comment by cameron-freshworks** [comment]
> No comments in addition to Cameron's. In future, Severin can also be added as a reviewer: he has done a lot of work on this site.

Good call!

Have added @severinbeauvais as reviewer (no pressure if you are busy though!)

File: enums/productCode.ts
Comment (dimak1) [reply enums/productCode.ts:6]: I wanted to keep the same way we display tiles on the page, and to display one 'My Asset' tile (instead of MHR tile and PPR tile) I added this product. Based on this product id we get all the content/links from the user-product-helper.

File: test/Dashboard.spec.js
Comment (severinbeauvais) [inline test/Dashboard.spec.js:20]: This is inconsistent with the use of `store` in our other UIs, where [store/index.ts](https://github.com/bcgov/business-edit-ui/blob/main/src/store/index.ts) creates the store, and both the app and the tests can use it.

Does this app create the store elsewhere, and you can't tap into that here?

File: utils/user-product-helper.ts
Comment (dimak1) [reply utils/user-product-helper.ts:93]: This unique `productCode` is used to get content for the tile from the `user-product-helper`.

`Products` is an array of objects that has a `ProductCode` in them. We need to iterate thru each product and check for the code. 

Screen Shot 2022-05-31 at 8 50 15 AM

File: test/Dashboard.spec.js
Comment (severinbeauvais) [inline test/Dashboard.spec.js:88]: The products are loading from where? Tests should not make actual API calls (those should be stubbed or mocked).

Timers in tests are always suspicious. If the code is simply async then you should be able to use nextTick() or flushPromises() ([example](https://github.com/bcgov/business-edit-ui/blob/223cc4f62e07396ab23ce73c98a65d2c2ee3d1dc/tests/unit/App.spec.ts#L355)).

File: test/UserProduct.spec.js
Comment (severinbeauvais) [inline test/UserProduct.spec.js:118]: There is "toBeNull()" that you can use here :)

File: components/UserProduct.vue
Comment (cameron-freshworks) [reply components/UserProduct.vue:62]: Sounds good 👍

File: enums/productCode.ts
Comment (cameron-freshworks) [reply enums/productCode.ts:6]: If it is a code unique to this UI perhaps leave a comment describing as such. I imagine these codes correlate to a product list that originates in Auth, so if we have something unique here we should highlight that.

File: utils/user-product-helper.ts
Comment (cameron-freshworks) [reply utils/user-product-helper.ts:93]: Ok fair enough, if you feel this is the best way i'm all for it 👍

File: test/UserProduct.spec.js
Comment (dimak1) [reply test/UserProduct.spec.js]: Refactored

File: enums/productCode.ts
Comment (dimak1) [reply enums/productCode.ts]: Refactored and removed this products alltogether

File: test/Dashboard.spec.js
Comment (dimak1) [reply test/Dashboard.spec.js:20]: For this app, we don't have the store set up this way, so I added it to the test.

**Comment by dimak1** [comment]
Thanks @cameron-freshworks and @severinbeauvais for your comments. I refactored and cleaned up products/utils/tests.

File: test/Dashboard.spec.js
Comment (severinbeauvais) [reply test/Dashboard.spec.js:20]: Thanks, that would be great.

File: test/Dashboard.spec.js
Comment (severinbeauvais) [reply test/Dashboard.spec.js:88]: OK, I see it now.

How about updating your test comment to refer to this specific timeout on the dashboard?

**Review by severinbeauvais — APPROVED** [review]
I'm hoping for the small fixes but approving in advance of them :)

File: pages/dashboard.vue
Comment (dimak1) [reply pages/dashboard.vue]: Sounds good, updated!

**Comment by dimak1** [comment]
@severinbeauvais thanks, I've made small fixes

File: test/Dashboard.spec.js
Comment (severinbeauvais) [reply test/Dashboard.spec.js:20]: I'm assuming the store works currently in the app, though I'm not sure how it's instantiated. Maybe you cannot tap into that mechanism in the tests; I dunno. If you can't find it then don't worry about it.

Comment (dimak1) [reply test/Dashboard.spec.js]: Updated to use globally defined VuexStore

File: enums/productCode.ts
Comment (cameron-freshworks) [reply enums/productCode.ts]: Awesome!

**Review by cameron-freshworks — APPROVED** [review]
Looks good, nice job!

**Comment by dimak1** [comment]
@severinbeauvais so PR is approved but checks failed. Also, I don't have permission to merge.

**Comment by cameron-freshworks** [comment]
> > @severinbeauvais so PR is approved but checks failed. Also, I don't have permission to merge.
> 
> When I click on the failed check details, it tells me someone needs to enter `/gcbrun`. Do you have permission to do that?
> 
> _Update: Because I typed that in (even in back-quotes), it looks like it's going to run this time. Next time, check if you can run it._

Looks like it might still require someone with permissions to approve the ci to run?

**Comment by dimak1** [comment]
Yea looks like it needs someone with the right access to run.

**Comment by dimak1** [comment]
Hi @thorwolpert looks like we don't have proper permissions to run (or merge) this PR.

**Comment by thorwolpert** [comment]
Temporary Url for review: https://yfthig-dev--pr-80-1qmtbgjz.web.app

**Comment by thorwolpert** [comment]
After UX approval, we'll merge this

**Comment by forgeuxGH5** [comment]
Design implementation looks good

**Pull request merged**
